File: backend/app/schemas/knowledge_base.py
import uuid
from datetime import date

from pydantic import BaseModel, ConfigDict

from app.models.base import LearningLevel


class ResourceRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    title: str
    provider: str
    level: LearningLevel
    is_free: bool
    description: str | None
    url: str
    last_verified_at: date


class GlossaryTermRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    term: str
    slug: str
    simple_explanation: str
    technical_explanation: str | None
    example: str | None

File: frontend/src/types/knowledgeBase.ts
export interface Resource {
  id: string
  title: string
  provider: string
  level: string
  is_free: boolean
  description: string | null
  url: string
  last_verified_at: string
}

export interface GlossaryTerm {
  id: string
  term: string
  slug: string
  simple_explanation: string
  technical_explanation: string | null
  example: string | null
}
